src/bika/lims/browser/fields/uidreferencefield.py

```diff
@@ -55,6 +54,17 @@ class UIDReferenceField(StringField):
 
     security = ClassSecurityInfo()
 
+    @property
+    def keep_backreferences(self):
+        """Returns whether this field must keep back references. Returns True
+        if the value for property relationship is None
+        """
+        if not isinstance(self.relationship, six.string_types):
+            return False
+        if not self.relationship:
+            return False
+        return True
```

Comment:
I don't understand this logic and the docstring.

How can it return `True` if the value is `None` when the first condition checks for strings only and the second condition for falsy values?

IMO the check should be smth. like this:

```python
@property
def keep_backreferences(self):
    relationship = getattr(self, "relationship", None)
    if relationship and isinstance(relationship, six.string_types):
        return True
    return False
```

But what should be now actually the desired behavior? As far as I understood it should not be linked unless explicitly set to a relationship key...

Reply:
Fixed with https://github.com/senaite/senaite.core/pull/2219/commits/414958130e93c1a38e358b864009d50f701d66f6 . Sometimes I overthink

Reply:
Okay, this means per default backrefs are not set. Is this currently only the case for client references in analysisrequest or for other references as well?
Would it make sense to purge the old references from these objects? I think about clients with thousands of samples linked as backreferences.

Reply:
> Okay, this means per default backrefs are not set. Is this currently only the case for client references in analysisrequest or for other references as well? 

This applies for all references, regardless of the source/target types. For this PR, I've searched for `get_backreferences` everywhere (add-ons included) to find the cases for which we use backreferences, but without the relationship being set actually. I then assigned the relationships manually to ensure that everything keeps working.

> Would it make sense to purge the old references from these objects? I think about clients with thousands of samples linked as backreferences.

Probably. I would expect a reduction of the size of the serialized objects. Is ok if I open another PR for this purge?